### PR TITLE
feat(NightDesign): inital night design basic presence

### PR DESCRIPTION
**Pull request by BCCSTeam**
## Description 
Repull request because of stupidness but with every requested change #7168
feat(NightDesign): inital night design basic presence

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
wWuhU
RmFKU

### CODE REVIEW

File: websites/N/NightDesign/metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.7",
+  "author": {
+    "id": "766213190332710913",
+    "name": "BassKatze#0001"
```
Comment (theusaf) [inline websites/N/NightDesign/metadata.json:5]:
```suggestion
    "name": "BassKatze"
```

File: websites/N/NightDesign/metadata.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "fabric",
+    "cloaks",
+    "capes",
+    "bandanas",
+    "shields",
+    "labymod",
+    "cosmeticsmod",
+    "morecosmetics"
+  ]
+}
```
Comment (theusaf) [inline websites/N/NightDesign/metadata.json:33]: This file is formatted incorrectly. Use `yarn format` to format it.

**Review by theusaf — CHANGES_REQUESTED** [review]
Note: You do not have to close and create a new PR when making new commits. They will automatically appear in the PR and re-run the checks.

File: websites/N/NightDesign/presence.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+			default:
+				presenceData.details = data;
+		}
+	}
+
+	if (!presenceData.details)
+		presenceData.details = document.title.split(" | ")[0] ?? "Night.design";
+	else {
+		presenceData.details =
+			strings.prefix + presenceData.details + strings.suffix;
```
Comment (DarkVillager) [inline websites/N/NightDesign/presence.ts:109]:
```suggestion
			`${strings.prefix}  ${presenceData.details} ${strings.suffix}`;
```

File: websites/N/NightDesign/presence.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+				if (pathSegments.length !== 2) {
+					presenceData.details = strings.resources;
+					continue;
+				}
+
+				presenceData.details = strings.resource.replace(
+					"$0",
+					textContent(
+						"div.page-header > div > div > div.col-md-4 > ul > li:nth-child(4) > a"
+					)
```
Comment (DarkVillager) [inline websites/N/NightDesign/presence.ts:65]: Simplify the selectors.
```suggestion
document.querySelector('[class="breadcrumb justify-content-end"]').lastChild?.textContent
```
or
```suggestion
textContent('[class="page-header-title"]')?.split(": ")[1]
```

File: websites/N/NightDesign/presence.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+				);
+				presenceData.state = strings.stateResources
+					.replace(
+						"$0",
+						toTitleCase(
+							textContent(
+								"div.page-content > div.row > div.col-md-7 > div:nth-child(2) > div.card-block > p > code:nth-child(1)"
```
Comment (DarkVillager) [inline websites/N/NightDesign/presence.ts:72]: Simplify selectors like this:
```suggestion
								'code[routerlink="/resources"]'
```

File: websites/N/NightDesign/presence.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+							)
+						)
+					)
+					.replace(
+						"$1",
+						textContent("div.card.mini-profile > div > div > h6 > span > span")
```
Comment (DarkVillager) [inline websites/N/NightDesign/presence.ts:78]: Also here simplify selectors like this:
```suggestion
						textContent('[class="d-inline-block"] > h6')
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/N/NightDesign/presence.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+						"div.page-header > div > div > div.col-md-4 > ul > li:nth-child(3)"
+					)
+				);
+				presenceData.state = toTitleCase(pathSegments.slice(-1)[0]);
+				break;
+
+			case "u":
+				presenceData.details = data.replace(
+					"$0",
+					textContent("div.page-header > div > div > div.col-md-8 > div > h5")
```
Comment (DarkVillager) [inline websites/N/NightDesign/presence.ts:95]:
```suggestion
					textContent('[class="page-header-title"]')
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/N/NightDesign/presence.ts [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+							)
+						)
+					)
+					.replace(
+						"$1",
+						textContent("div.card.mini-profile > div > div > h6 > span > span")
+					);
+				break;
+
+			case "server":
```
Comment (DarkVillager) [inline websites/N/NightDesign/presence.ts:82]: Cant seem to access this page?

Comment (BCCSTeam) [reply websites/N/NightDesign/presence.ts:82]: not accessible yet

**Review by Bas950 — APPROVED** [review]

**Comment by Bas950** [comment]
Low quality presence, support more pages

**Pull request closed without merging**